front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -208,6 +221,51 @@ export const usePathfinding = (
     }
   }, [origin?.id, destination?.id, rollingStock]);
 
+  const handleInvalidPathItems = (
+    pathfindingResult: PathfindingResult,
+    pathSteps: (PathStep | null)[],
+    dispatch: AppDispatch,
+    pathfindingDispatch: React.Dispatch<PathfindingAction>,
+    t: TFunction
+  ) => {
+    if (
+      pathfindingResult.status === 'failure' &&
+      pathfindingResult.failed_status === 'pathfinding_input_error' &&
+      pathfindingResult.error_type === 'invalid_path_items'
+    ) {
+      //TODO: we currently only handle invalid pathSteps with trigram. We will have to do it for trackOffset, opId and uic too.
+      const invalidPathItems = pathfindingResult.items
+        .map((item) => {
+          if ('trigram' in item.path_item) {
+            return item.path_item.trigram;
+          }
+          return null;
+        })
+        .filter((trigram): trigram is string => trigram !== null);
+      if (invalidPathItems.length > 0) {
+        const updatedPathSteps = pathSteps.map((step) => {
+          if (step && 'trigram' in step && invalidPathItems.includes(step.trigram)) {
+            return { ...step, isInvalid: true };
+          }
+          return step;
+        });
+
+        dispatch(updatePathSteps({ pathSteps: updatedPathSteps }));
+        pathfindingDispatch({ type: 'PATHFINDING_FINISHED' });
+        pathfindingDispatch({ type: 'PATHFINDING_PARAM_CHANGED' });
+      } else {
+        dispatch(
+          setFailure(
+            castErrorToFailure(pathfindingResult, {
+              name: t('pathfinding_failed'),
+              message: t('missing_pathSteps'),
+            })
+          )
```

Comment:
You can directly setFailure with a name and a message :)
```suggestion
         setFailure({
            name: t('pathfinding_failed'),
            message: t('missing_pathSteps'),
          })
```

Reply:
done